### PR TITLE
doc: add challenge-of-microservices blog

### PR DESCRIPTION
**Pull request by coltmcnealy-lh**
- Adds "Challenges of Microservices" blog
- Adds short 0.10.0 release blog
- Updates docusaurus to 3.5.1
- Organizes blog authors and tags

### CODE REVIEW

**Review by mijailr — APPROVED** [review]

File: docs/blog/2024-07-12-0.10-release.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+hide_table_of_contents: false
+---
+
+The `0.10` release brings with it significant performance and reliability improvements. <!-- truncate -->
+
+## New Features
+
+### `lhctl` Binaries and Release Notes
+
+The `0.10.0` release comes with a new [Release Page](https://github.com/littlehorse-enterprises/littlehorse/releases), including artifacts with `lhctl` built for ARM, Intel, and Windows.
```
Comment (mitchell-h) [inline docs/blog/2024-07-12-0.10-release.md:18]: How about "Including releases of `lhctl` built for ARM, Intel, and Windows."

Artifacts is a heavy word, "releases" is more easily understood.

File: docs/blog/2024-07-12-0.10-release.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+With this feature, a user can fix a buggy Task Worker implementation and then restart a failed `WfRun` and get it to execute the failed `TaskRun` again via:
+
+```
+lhctl rescue <wfRunId> <threadRunNumber>
+```
+
+### mTLS Principals
+
+Previously, only listeners of the type `OAUTH` supported `Principal`s. The `Principal` ID was determined by the OAuth Client ID or User Id. Release `0.10` introduces the ability to infer a `Principal` on an `MTLS` listener, where the `Principal` ID comes from the Common Name on the client certificate.
```
Comment (mitchell-h) [inline docs/blog/2024-07-12-0.10-release.md:36]: Add a summary.  Something like "This enables users that standardize on SSL authentication to continue that scheme into Littlehorse."

File: docs/blog/2024-08-27-challenges-of-microservices.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+While systems like Apache Kafka have clear leaders (for example, the _Controller_ may re-assign partition leadership if the cluster becomes too imbalanced), in a microservice-based system there is no central leader to ensure that the chips fall correctly. 
+
+You can think of our e-commerce microservice flow as a line of dominoes falling. Once the process starts, no one entity is responsible for ensuring its completion. The business workflow moves from `orders` to `inventory` to `payments` and so on. If `payments` fails for some reason (perhaps a network partition makes the Stripe API unavailable), then it's quite possible that the `shipping` service never finds out about the workflow.
+
+However, in real life such outcomes are not acceptable. This means that every single player in the system must:
+
+1. Have built-in reliability mechanisms.
+2. Understand the preceding and subsequent steps of the business process to route traffic.
+
```
Comment (mitchell-h) [inline docs/blog/2024-08-27-challenges-of-microservices.md:86]: What is the problem with adding these to each of these?  Why would having to code in #1 and #2 be bad?

Suggestion:  Implementing the above slows down development, more tightly couples one services to another, increases dependancies, and negates the advantages of a microservices archiecture.

File: docs/blog/2024-08-27-challenges-of-microservices.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+So far, we have established that there are many players involved in a business process, yet there's no one player in charge of ensuring that the trains run on time. This yields three problems:
+
+1. **Reliability** in the face of infrastructure failures.
+2. **Observability** to enable system optimization and debugging.
+3. **Coupling** of microservices to each other makes it hard to modify the system in response to new business requirements.
+
+### Reliability and Correctness
+
+Processing orders is a mission-critical use-case. This means that orders should always complete and never be dropped (for example, we shouldn ever charge the customer's credit card and not ship the product to them).
```
Comment (mitchell-h) [inline docs/blog/2024-08-27-challenges-of-microservices.md:97]: s/shouln/should not

File: docs/blog/2024-08-27-challenges-of-microservices.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+However, asynchronous processing such as that which I outlined above is prone to failures. For example, if you chain microservices together with direct RPC calls, a single network partition can cause an order to get stuck. Even with a reliable message broker such as Apache Kafka or AWS SQS sitting between your microservices, a write to the message broker could fail _after_ the payment went through, still resulting in a stuck order.
+
+Just as communication _between_ microservices can fail, the actions performed _by_each microservice can also fail. If the Stripe API is down, or if the credit card is invalid, we can't just stop processing the order there! We must notify the customer of what went wrong and also release the inventory that we reserved.
```
Comment (mitchell-h) [inline docs/blog/2024-08-27-challenges-of-microservices.md:101]: Highlight the call out to external systems.  
Suggestion:
Just as communication _between_ microservices can fail, the actions performed _by_each microservice can also fail.  In many cases actions performed by a microservice must be performed and the resulting return determines what steps to take next.  External systems, due to the additional steps involved in communicating them with, are often more prone for failure.  For example if the Stripe API is down, or if the credit card is invalid, we can't just stop processing the order there! We must notify the customer of what went wrong and also release the inventory that we reserved.

File: docs/blog/2024-08-27-challenges-of-microservices.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+### Observability
+
+The second problem with microservices is that once a process instance has started (i.e. the dominoes are falling), it is very difficult to observe what happens between steps 2 through 10. This means that multi-step processes with performance issues are hard to optimize, as there are many microservices which could be the bottleneck and it's hard to know which. Even worse, when a customer complains about a "stuck order," it is difficult to find the point of failure.
+
+As a result, microservice engineers spend time and money:
+* Slogging through logs on DataDog
+* Implementing complex distributed tracing such as Zipkin, Jaeger, or Kiali
+* Saving the state of each process instance (in our case, the `order`) in a DB just for visibility purposes at every step.
+
```
Comment (mitchell-h) [inline docs/blog/2024-08-27-challenges-of-microservices.md:121]: Add: 
* Coordinating with other teams to manually understand and debug workflows

**Review by Snarr — APPROVED** [review]

**Pull request merged**